lib/presentation/widgets/custom_icon_button.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:trip_tally/presentation/theme/app_colors.dart';
+
+import '../theme/app_dimensions.dart';
+
+class CustomIconButton extends StatelessWidget {
+  const CustomIconButton({
+    super.key,
+    required this.icon,
+    required this.onTap,
+  });
+
+  final String icon;
```

Comment:
You can also change the `icon` into `svg` so when someone use it will instantly know what to parse here